core/shortcut_items.ts

```diff
@@ -69,7 +69,8 @@ export function registerDelete() {
     name: names.DELETE,
     preconditionFn(workspace) {
       const selected = common.getSelected();
-      return !workspace.options.readOnly && selected && selected.isDeletable();
+      return !!(
+          !workspace.options.readOnly && selected && selected.isDeletable());
```

Comment:
Could you explain why this is necessary? I'm not groking it.

Reply:
It wasn't clear to me either, upon further inspection it's because of the `&& selected` and JS's delightful type coercion behavior - `true && null evaluates` to `null`, so it was possible for this to return `boolean|null`.

Reply:
Goootcha gotcha. I kind of prefer `!workspace.options.readOnly && !!selected && selected.isDeletable()` because that makes it clear which part we are coercing to a boolean. But feel free to merge as is (i.e. go with whatever you prefer).